src/test/java/jenkins/plugins/logstash/LogstashWriterTest.java

```diff
@@ -0,0 +1,299 @@
+    LogstashWriter writer = createLogstashWriter(mockBuild, errorBuffer, "http://my-jenkins-url", null, null);
+    assertTrue("Connection not broken", writer.isConnectionBroken());
+
+    String msg = "test";
+    errorBuffer.reset();
+
+    // Unit under test
+    writer.write(msg);
+
+    // Verify results
+    assertEquals("Results don't match", "", errorBuffer.toString());
+    assertTrue("Connection not broken", writer.isConnectionBroken());
+  }
+
+  @Test
+  public void writeBuildLogSuccessNoDao() throws Exception {
+    LogstashWriter writer = createLogstashWriter(mockBuild, errorBuffer, "http://my-jenkins-url", null, null);
+    assertTrue("Connection not broken", writer.isConnectionBroken());
+
+    String msg = "test";
+    errorBuffer.reset();
+
+    // Unit under test
+    writer.writeBuildLog(3);
+
+    // Verify results
+    assertEquals("Results don't match", "", errorBuffer.toString());
+    assertTrue("Connection not broken", writer.isConnectionBroken());
+  }
+
+  @Test
+  public void writeSuccess() throws Exception {
+    LogstashWriter writer = createLogstashWriter(mockBuild, errorBuffer, "http://my-jenkins-url", mockDao, mockBuildData);
+    String msg = "test";
+    errorBuffer.reset();
+
+    // Unit under test
+    writer.write(msg);
+
+    // Verify results
+    // No error output
+    assertEquals("Results don't match", "", errorBuffer.toString());
+
+    verify(mockDao).buildPayload(Matchers.eq(mockBuildData), Matchers.eq("http://my-jenkins-url"), Matchers.anyListOf(String.class));
+    verify(mockDao).push("{\"data\":{},\"message\":[\"test\"],\"source\":\"jenkins\",\"source_host\":\"http://my-jenkins-url\",\"@version\":1}");
+  }
+
+  @Test
+  public void writeBuildLogSuccess() throws Exception {
+    LogstashWriter writer = createLogstashWriter(mockBuild, errorBuffer, "http://my-jenkins-url", mockDao, mockBuildData);
+    errorBuffer.reset();
+
+    // Unit under test
+    writer.writeBuildLog(3);
+
+    // Verify results
+    // No error output
+    assertEquals("Results don't match", "", errorBuffer.toString());
+    verify(mockBuild).getLog(3);
+
+    verify(mockDao).buildPayload(Matchers.eq(mockBuildData), Matchers.eq("http://my-jenkins-url"), Matchers.anyListOf(String.class));
+    verify(mockDao).push("{\"data\":{},\"message\":[\"test\"],\"source\":\"jenkins\",\"source_host\":\"http://my-jenkins-url\",\"@version\":1}");
+  }
+
+  @Test
+  public void writeSuccessConnectionBroken() throws Exception {
+    Mockito.doNothing().doThrow(new IOException("BOOM!")).doNothing().when(mockDao).push(anyString());
+    LogstashWriter los = createLogstashWriter(mockBuild, errorBuffer, "http://my-jenkins-url", mockDao, mockBuildData);
+
+
+    String msg = "test";
+    String exMessage = "[logstash-plugin]: Failed to send log data to REDIS:localhost:8080.\n" +
+      "[logstash-plugin]: No Further logs will be sent to localhost:8080.\n" +
+      "java.io.IOException: BOOM!\n";
```

Comment:
You don't really need the trailing newline character, since you are testing if this is a substring of the result.